src/Apps/Settings/Routes/Insights/Components/InsightsMedianSalePrice.tsx

```diff
@@ -0,0 +1,106 @@
+import { Box, Flex, Text } from "@artsy/palette"
+import { EntityHeaderArtistFragmentContainer } from "Components/EntityHeaders/EntityHeaderArtist"
+import { groupBy } from "lodash"
+import { createFragmentContainer, graphql } from "react-relay"
+import { extractNodes } from "Utils/extractNodes"
+import { InsightsMedianSalePrice_me$data } from "__generated__/InsightsMedianSalePrice_me.graphql"
+
+interface InsightsMedianSalePriceProps {
+  me: InsightsMedianSalePrice_me$data
+}
+
+const InsightsMedianSalePrice: React.FC<InsightsMedianSalePriceProps> = ({
+  me,
+}) => {
+  const medianSalePrices = extractNodes(me.medianSalePrices)
+
+  if (!medianSalePrices.length) {
+    return null
+  }
+
+  const groupedMedianSalePrices = Object.values(
+    groupBy(medianSalePrices, msp => msp?.artist?.internalID)
+  )
+
+  return (
+    <Box>
+      <Text variant={["sm-display", "md"]}>
+        Median Auction Price in the Last 3 Years
+      </Text>
+
+      {groupedMedianSalePrices.map(msps => {
```

Comment:
**question:** What does `msps` stand for?

Reply:
Good question! I used `msps` as a short for MarketSalePrices instead of `marketSalePrices` :)

Reply:
Personally, I like to avoid abbreviations as much as possible in the code to make it easier to read and understand ([some random article on the internet about the topic](https://blog.boot.dev/clean-code/naming-variables/#8-dont-use-abbreviations-or-acronyms-without-sufficient-context)).

Reply:
same here, I wouldn't shy away from using a long descriptive name